README.md

```diff
@@ -142,6 +142,135 @@ const Person = memo(props => {
+For simple cases it is recommended to use `React.setState`, but if your component requires complex view models, consider creating a local mobx store by using this hook.
+
+If the returned value is a plain object, it will be automatically be passed through `observable`, turning fields into observable properties, and `get` based property accessors in computed values, and functions in bound actions.
+
+If new class instances are returned from the initializer, they will be kept as is. Quick example:
+
+```typescript
+function Counter() {
+    const store = useLocalStore(() => ({
+        count: 0,
+        inc() {
+            this.count += 1
+        }
+    }))
+
+    return useObserver(() => (
+        <div>
+            Count: {store.count}
+            <button onClick={store.inc}>Increment</button>
+        </div>
+    ))
+}
+```
+
+It is important to realize that the store is created only once! It is not possible to specify dependencies to force re-creation, _nor should you directly be referring to props for the initializer function_, as changes in those won't propagate.
+
+Instead, if your store needs to refer to props (or `useState` based local state), the `useLocalStore` should be combined with the `useAsObservableSource` hook, see below.
+
+### `useAsObservableSource<T>(state: T): T`
+
+The `useAsObservableSource` hook can be used to turn any set of values into an observable object that has a stable reference (the same object is returned every time from the hook).
+The goal of this hook is to trap React primitives such as props or state into a local, observable object
+so that the `store` initializer can safely refer to it, and get notified if any of the values change.
+
+Example:
+
+```typescript
+function Counter({ multiplier }) {
+    const observableProps = useAsObservableSource({ multiplier })
+    const store = useLocalStore(() => ({
+        count: 0,
+        get multiplied() {
+            return observableProps.multiplier * this.count
+        },
+        inc() {
+            store.count += 1
+        }
+    }))
+
+    return useObserver(() => (
+        <div>
+            Multiplied count: {store.multiplied}
+            <button onClick={store.inc}>Increment</button>
+        </div>
+    ))
+}
+```
+
+In the above example, any change to `multiplier` prop will show up in the `observableProps` observable object, and be picked up by the `store`.
+
+Reading properties for the returned observable object, should always be done _inside_ the store initializer (respectively side-effect, see below), to avoid values being trapped in the closure.
+In other words: _the return value of `useAsObservableSource` should never be deconstructed! So, don't write: `const {multiplier} = useAsObservableSource({ multiplier })`!_
+
+Note that the value passed to `useAsObservableSource` should always be an object, and is made only shallowly observable.
```

Comment:
This is fairly important to be just a note imo. Honestly, I am unsure myself what would happen with array value on that passed object. Is it going to be observable on itself? I am a bit 🤧 so I not thinking clearly right now...

Reply:
I think it can be a note, because in all other case we will be throwing an exception immediately anyway :P